Lib/test/test_concurrent_futures/test_deadlock.py

```diff
@@ -241,6 +243,65 @@ def test_crash_big_data(self):
 
         executor.shutdown(wait=True)
 
+    def test_gh105829_should_not_deadlock_if_wakeup_pipe_full(self):
+        # Issue #105829: The _ExecutorManagerThread wakeup pipe could
+        # fill up and block. See: https://github.com/python/cpython/issues/105829
+
+        # Lots of cargo culting while writing this test, apologies if
+        # something is really stupid...
+
+        self.executor.shutdown(wait=True)
+
+        if not hasattr(signal, 'alarm'):
+            raise unittest.SkipTest(
+                "Tested platform does not support the alarm signal")
+
+        def timeout(_signum, _frame):
+            raise RuntimeError("timed out while submitting jobs?")
```

Comment:
For easier debugging and making sure we observe the right deadlock, I usually use `faulhandler`, and check that the executor manager thread is waiting on the shutdown lock.
```suggestion
        def timeout(_signum, _frame):
            import faulthandler
            faulthandler.dump_traceback()

            raise RuntimeError("timed out while submitting jobs?")
```